### ISSUE TITLE
Add the ability to configure the slowDB query threshold

### ISSUE THREAD

**Issue by AndrewGrossman** [getsentry/sentry#64411]
### Problem Statement

See the original issue at https://github.com/getsentry/sentry/issues/47020, which became stale and was auto-closed. It's open because I wanted to +1/comment on it and someone in support reopened it. However, it's still locked from external input.

My team has an application with two types of queries: 
1. Typical activities associated with most user interactions
2. Report-type queries that process a larger amount of data

For us, those Type-2 reporting queries take a few seconds. That's perfectly acceptable for us and the particular context. If those jumped from 10 seconds to 60, we might want to know. As things stand, those queries show up as noise in our slow query monitoring. But we don't want to simply turn it off; we do want to know if both types of become problematic.

### Solution Brainstorm

If it weren't configurable up front, another option could be a slow query-specific `archive` setting. That is, Archive _until_ query time goes above _x_ threshold.

### Product Area

Issues

**Comment by gggritso** [comment]
@AndrewGrossman are you looking for a simple project-wide threshold? If so, you can configure the threshold for anywhere between 100ms and 10s in the "Performance" section of project settings, as described in [the product docs](https://docs.sentry.io/product/issues/issue-details/performance-issues/slow-db-queries/#detection-criteria). Does that solve your problem?

**Comment by AndrewGrossman** [comment]
@gggritso Not quite. The different use case sets from the description are within the same codebase and project.

**Comment by gggritso** [comment]
@AndrewGrossman understood, thanks for raising this! This is an interesting use-case, we don't have a good answer to this right now. I'm going to keep this open, hopefully we'll have a chance to address this kind of situation in an upcoming iteration.

A few more notes from me, for posterity:

- if we find a good way to integrate Metrics with span durations, we can allow people to set custom thresholds for alerts on query durations, which would obviate Slow Query issues and might make for a cleaner overall solution
- a "Regressed Queries" section in the Queries module powered by the same backend as regressed functions and transactions might answer a lot of these questions
- Sentry has a similar situation where some analytics queries are very slow, and that's fine. Splitting this by SDK is too annoying and _lots_ of apps have this kind of bimodal behaviour
- we chose, back in the day, to detect performance issues on the backend in the service rather than the SDK. One downside is that users aren't able to granularly control detection in the SDKs config of their apps, though IMO SDK configuration isn't a very friendly solution to this kind of problem

**Comment by thatsanicehat** [comment]
I would like to add my desire to be able to more granularly configure this as well. We have asynchronous queries that run every 15 minutes that take several seconds and have caused thousands of alerts as a result. We know they're expensive, which is why we run them async. 

I'm not _too_ worried about them eating much into our ~400K error threshold quota, and could certainly just archive them forever. 

However, if they go from several seconds to, say, a minute, that would indicate a serious issue that we should look into. We also can build our own logic around known queries and manually send exceptions to Sentry if we'd like, but that sort of manual setup is cumbersome and would require either an external configuration tool to adjust or code delivery, which isn't ideal. 

The ability to adjust the "long database query" threshold on a per-tag basis would be fantastic.

**Comment by gggritso** [comment]
@thatsanicehat thanks for raising! We're working on related parts of the product in the near future, and hopefully we'll have a better answer for this problem soon.
